app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -79,4 +104,14 @@ def add_disk(vmcs, disk, controller, new_dev_key)
       end
     end
   end
+
+  private
+
+  def disks
```

Comment:
And this would be `devices` since `config.hardware.device` include all the devices not just disks

Also we should be caching this result